setuptools/dist.py

```diff
@@ -999,6 +979,18 @@ def run_command(self, command) -> None:
         super().run_command(command)
 
 
+@functools.cache
+def _setuptools_commands() -> set[str]:
+    try:
+        # Use older API for importlib.metadata compatibility
+        entry_points = metadata.distribution('setuptools').entry_points
+        eps = (ep.name for ep in entry_points)
+    except metadata.PackageNotFoundError:
+        # during bootstrapping, distribution doesn't exist
+        return set(distutils.command.__all__)
+    return {*distutils.command.__all__, *eps}
+
+
```

Comment:
I moved this function out of the class to avoid memory leaks (otherwise it would keep a reference to `self`).